llvm/utils/TableGen/VTEmitter.cpp

```diff
@@ -30,7 +30,8 @@ class VTEmitter {
 } // End anonymous namespace.
 
 void VTEmitter::run(raw_ostream &OS) {
-  emitSourceFileHeader("ValueTypes Source Fragment", OS);
+  emitSourceFileHeader("ValueTypes Source Fragment", OS,
+                       "llvm/include/llvm/CodeGen/ValueTypes.td");
```

Comment:
I don't like such a hardcoded file name.
Could it be generated from given source file name? (In TableGen or CMakeLists)

Reply:
You can get the input file from RecordKeeper::getInputFileName()

Reply:
Addressed, thanks!